spec/std/http/request_spec.cr

```diff
@@ -138,6 +154,14 @@ module HTTP
       request.headers.should eq({"Host" => "host.example.org"})
     end
 
+    it "parses empty string (EOF), returns nil" do
+      Request.from_io(IO::Memory.new("")).should be_nil
+    end
+
+    it "parses empty string (EOF), returns nil (no peek)" do
+      Request.from_io(EmptyIO.new).should be_nil
```

Comment:
This is fine, but it's actually the only spec explicitly for a non-peek IO. We should probably run every request spec against both a peekable and non-peekable IO (not in this PR though).